README.md

```diff
@@ -7,34 +7,45 @@ Initialize and update submodules:
 git submodule update --init --recursive
 ```
 
-Run the following code to setup the virtual environment, add the python files in src to python's
-import path, then run the venv
+## Download Datasets
 
-```
-python3 -m venv venv
+You can download all the datasets we use in the benchmark using the [download\_all.py](/scripts/download_all.py) script we provide.
+
+The [download\_all.py](/scripts/download_all.py) script will download all datasets into the correct directories **with** the specified names, concentrate multi-file datasets together into a single file, and generate any modified version of the dataset needed for tools like Presto \+ CLP.
+
+## Docker Environments
+
+Benchmarks are executed inside Docker containers to ensure reproducible, isolated test environments
+with controlled resource limits.
 
-echo "$(pwd)" > $(find venv/lib -maxdepth 1 -mindepth 1 -type d)/site-packages/project_root.pth
+### Build All Docker Images
 
-. venv/bin/activate
+To build all Docker images concurrently:
 
-pip3 install -r requirements.txt
+```shell
+task docker-images:build
 ```
 
-## Download Datasets
+### Build a Single Docker Image
 
-You can download all the datasets we use in the benchmark using the [download\_all.py](/scripts/download_all.py) script we provide.
+To build a specific image for a given service engine:
 
-The [download\_all.py](/scripts/download_all.py) script will download all datasets into the correct directories **with** the specified names, concentrate multi-file datasets together into a single file, and generate any modified version of the dataset needed for tools like Presto \+ CLP.
+```shell
+uv run src/log_archival_bench/scripts/docker_images/build.py --service-name <service_name>
+```
+
+Each valid service (e.g. `presto`, `clickhouse`, `elasticsearch`) has its own dedicated Docker
+image, built by the `log-archival-bench` repository, for use in benchmark testing.
 
 ## Run Everything
 
 Follow the instructions above to set up your virtual environment.
 
 Stay in the [Log Archival Bench](/) directory and run [scripts/benchall.py](/scripts/benchall.py). This script runs the tools \+ parameters in its "benchmarks" variable across all datasets under [data/](/data).
 
-## Run One Tool
+## Run One Benchmark Engine
 
-Execute `./assets/{tool name}/main.py {path to <dataset name>.log}` to run ingestion and search on that dataset.
+Execute `./assets/{engine_name}/main.py {path to <dataset_name>.log}` to run ingestion and search on that dataset.
```

Comment:
maybe no need to touch this block for now.
we can also defer README fixing once we've moved everything into `src/log_archival_bench`